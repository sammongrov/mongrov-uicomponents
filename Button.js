import { Button as RNEButton } from 'react-native-elements';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Button extends Component {
  render() {
    const { title, buttonStyle, containerStyle, onPress } = this.props;
    return (
      <RNEButton
        title={title}
        buttonStyle={buttonStyle}
        containerStyle={containerStyle}
        onPress={onPress}
      />
    );
  }
}

Button.propTypes = {
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  buttonStyle: PropTypes.instanceOf(Object),
  containerStyle: PropTypes.instanceOf(Object),
};

Button.defaultProps = {
  buttonStyle: {},
  containerStyle: {},
};
