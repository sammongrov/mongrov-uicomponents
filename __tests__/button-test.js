import React from 'react';
import renderer from 'react-test-renderer';
import { Alert } from 'react-native';
import Button from '../Button';

const onPress = () => Alert.alert('press');

it('Button renders correctly', () => {
  const tree = renderer
    .create(<Button text="hello" activeOpacity={0.8} onPress={onPress} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
