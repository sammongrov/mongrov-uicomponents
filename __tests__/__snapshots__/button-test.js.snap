// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      false,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderRadius": 3,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {},
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            undefined,
            false,
            false,
          ]
        }
      >
        Welcome to
React Native Elements
      </Text>
    </View>
  </View>
</View>
`;
